src/utils/callApi.js

```diff
@@ -0,0 +1,50 @@
+import axios from "axios";
+import { API_URL } from ".";
+export const callApi = async (
+  method,
+  endPoint,
+  isProtected = false,
+  data = {}
+) => {
+  const token = JSON.parse(localStorage?.getItem("user"))?.token;
+  endPoint = `${API_URL}/${endPoint}`;
+  switch (method) {
+    case "get": {
+      return await axios.get(
+        endPoint,
+        isProtected && {
+          headers: { authorization: token },
```

Comment:
Suggestion: You can defer adding the headers part to axios interceptors,. That way you don't need to pass headers on each request